### ISSUE TITLE
Request: globally set some attributes (to reduce watch count)

### ISSUE THREAD

**Issue by robmaurer** [angular-ui-tree/angular-ui-tree#1046]
**Do you want to request a *feature* or report a *bug*?**
Feature request

**What is the current behavior?**
Each tree node adds watches for values of `expandOnHover` and `scrollContainer`.

**What is the expected behavior?**
Large trees need an option to set these values globally for the tree (a configuration) and store them in memory (not watch the DOM for them on each node).

**What is the motivation / use case for changing the behavior?**
In trees having hundreds of nodes, would reduce number of watches on DOM, for performance.

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**
Not doing `scope.$watch(attrs.expandOnHover, function (val) {})` and not doing `attrs.$observe('scrollContainer', function(val) {})` reduces DOM watch load by two per node 
(as measured by this methodology: https://medium.com/@kentcdodds/counting-angularjs-watchers-11c5134dc2ef

While the ability to set `expandOnHover` by node is essential for some use-cases, an optimization would be an opt-in configuration to set it for the tree and remove the $watch. With `scrollContainer`, I can imagine no use-case that would break by set-and-forget.

**Comment by dgutride** [comment]
@Voles - is there any chance if we submit a PR that it will be taken in?

**Comment by Voles** [comment]
@dgutride not really, to be honest. Sorry, this project has been unmaintained for a couple of years now.

**Comment by jtrout** [comment]
> @Voles - is there any chance if we submit a PR that it will be taken in?

Do you have a branch or fork that addresses this? We're still using the control in a legacy project and the performance with big data is pretty brutal, but otherwise the control is exactly what we need. Or is the fix here just as straightforward as it sounds?

**Comment by maykon-oliveira** [comment]
I'd like to fork and continue with the support.
